### ISSUE TITLE
[HOLD for payment 2023-12-29] [$500] Room - In admin room, when copy to clipboard, the room name is not shown as link

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#31568]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.1-5
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

**Issue found when executing PR** https://github.com/Expensify/App/pull/30898
## Action Performed:
1. Launch app
2. Tap start chat--- Room
3. Tap header---Invite a user to the room
4. In room page, long press the invite message and select copy to clipboard
5. Paste it in compose box and send the message
6. Note the copied message is displayed correctly
7. Navigate to LHN
8. Tap on admin room
9. Long press the invite message and select copy to clipboard
10. Paste it in compose box and send the message

## Expected Result:
In admin room, when copy to clipboard, paste and send the invite message, the room name must be shown as link

## Actual Result:
In admin room, when copy to clipboard, paste and send the invite message, the room name is not shown as link

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/1a089d40-3a51-4c19-9852-c591db71ea6b

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a7ae55acdd8221b2</li>
        <li>Upwork Job ID: 1726667353323319296</li>
        <li>Last Price Increase: 2023-11-20</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 27850800</li>
<li>tienifr | Contributor | 27850801</li>
</ul></li>
    </ul>
</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
In admin room, room name is not copied as link when copying invite message.
### What is the root cause of that problem?
We are just only including the room name as a text (not link) to the clipboard as in here:
https://github.com/Expensify/App/blob/b4f001de6b3fe232a0f5aee5316b18557b740d86/src/libs/ReportUtils.js#L4213-L4219
### What changes do you think we should make in order to solve the problem?

We should extract the room link from `message.html` and parse it to markdown (so that it appears as a markdown to the compose) and copy to clipboard text
Change this 
https://github.com/Expensify/App/blob/b4f001de6b3fe232a0f5aee5316b18557b740d86/src/libs/ReportUtils.js#L4219
to 
```
        const messageHtml = lodashGet(_.last(lodashGet(reportAction, 'message', [{}])), 'html');
        const roomLink = /<\/mention-user> to (<a href=[^>]*>#[^<]*<\/a>)/.exec(messageHtml)[1];
        const parser = new ExpensiMark();
        message += `${preposition} ${parser.htmlToMarkdown(roomLink)}`;
```
POC

https://github.com/Expensify/App/assets/38649957/ad8b8022-e2e4-4fa5-b5cf-8a006018acfb



### What alternative solutions did you explore? (Optional)

**Comment by shubham1206agra** [comment]
Thank you, both @FitseTLT and @tienifr, for your proposals.

Since both are proposing similar solutions and since @tienifr's [proposal](https://github.com/Expensify/App/issues/31568#issuecomment-1819605751) came first. I think we should go with that.

🎀 👀 🎀 C+ reviewed

**Comment by luacmartins** [comment]
@shubham1206agra can you check if this is a regression of https://github.com/Expensify/App/pull/30898? If so, we should have @dukenv0307 work on a fix

**Comment by shubham1206agra** [comment]
I don't think that this is a regression. I think they only checked for the same text, which is reflected in their QA test.

If we want to create a link here, which is a design choice, then we can treat it as a separate issue.

**Comment by shubham1206agra** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
* [x]  [@shubham1206agra] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/30898
* [x]  [@shubham1206agra] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/30898/files#r1440628263
* [x]  [@shubham1206agra] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not required
* [x]  [@shubham1206agra] Determine if we should create a regression test for this bug. No as this should be made more clear from design perspective while implementing https://github.com/Expensify/App/pull/30898
* [x]  [@shubham1206agra] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
* [ ]  [@MitchExpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MitchExpensify** [comment]
Sorry there is a lot going on here and I'm having trouble identifying when the PR specifically related to this bug hit production, can you help me confirm that please @luacmartins ?

**Comment by shubham1206agra** [comment]
@luacmartins Bump on the above ^

**Comment by luacmartins** [comment]
Yes, [this was the PR](https://github.com/Expensify/App/pull/32161) with the fix and it hit [prod on Dec 21](https://github.com/Expensify/App/pull/32161#issuecomment-1867252780)

**Comment by MitchExpensify** [comment]
Thank you! Paid and contracts ended, @tienifr @shubham1206agra 👍 Thanks for your patience!
